ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/HiveRelOptMaterializationValidator.java

```diff
@@ -308,24 +318,19 @@ public boolean isValidForQueryCaching() {
     return resultCacheInvalidReason == null;
   }
 
-  public String getAutomaticRewritingInvalidReason() {
-    return automaticRewritingInvalidReason;
-  }
-
-  public void setAutomaticRewritingInvalidReason(String automaticRewritingInvalidReason) {
-    if (isValidForAutomaticRewriting()) {
-      this.automaticRewritingInvalidReason = automaticRewritingInvalidReason;
-    }
+  public AutomaticRewritingValidationResult getAutomaticRewritingValidationResult() {
+    return automaticRewritingValidationResult;
   }
 
-  public void setAutomaticRewritingInvalidReason(RelNode node) {
+  public void unsupportedByCalciteRewrite(String sqlPartType, String sqlPart) {
     if (isValidForAutomaticRewriting()) {
```

Comment:
There are two types of failures:
1. disable both rewrite algs.
2. allow text base only
If type 1) is found we can stop traversing the tree.
If type 2) is found we record it if this is the first one from this type but we have to continue traversing because there might be a type 1).

Actually we print only one failure and if there are more than one from type 2) it doesn't matter which one is printed. I kept the logic avoid too many golden file changes.